js/tuxexchange.js

```diff
@@ -0,0 +1,567 @@
+            const tickerSymbol = tickerResult['symbol'];
+            if (tickerSymbol === symbol) {
+                return tickerResult;
+            }
+        }
+        throw new BadRequest (this.id + ' no ticker found for symbol provided');
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        await this.loadMarkets ();
+        let res = await this.publicGet ({ 'method': 'getticker' });
+        const tickersData = this.parseJson (res);
+        const tickerIds = Object.keys (tickersData);
+        const result = {};
+        for (let i = 0; i < tickerIds.length; i++) {
+            const id = tickerIds[i];
+            const tickerResult = this.parseTicker (id, tickersData[id]);
+            const symbol = tickerResult['symbol'];
+            result[symbol] = tickerResult;
+        }
+        return result;
+    }
+
+    parseTicker (marketId, tickerData) {
+        const timestamp = this.nonce (); // TODO: hopefully this approximation is sufficient
+        const market = this.findMarket (marketId);
+        // ALGO: all the values here check out, but there are a lot of undefineds
+        return {
+            'symbol': market['symbol'],
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'close': this.safeFloat (tickerData, 'last'),
+            'last': this.safeFloat (tickerData, 'last'),
+            'high': this.safeFloat (tickerData, 'high24hr'),
+            'low': this.safeFloat (tickerData, 'low24hr'),
+            'percentage': this.safeFloat (tickerData, 'percentChange'),
+            'baseVolume': this.safeFloat (tickerData, 'baseVolume'),
+            'quoteVolume': this.safeFloat (tickerData, 'quoteVolume'),
+            'bid': this.safeFloat (tickerData, 'highestBid'),
+            'ask': this.safeFloat (tickerData, 'lowestAsk'),
+            'bidVolume': undefined,
+            'askVolume': undefined,
+            'vwap': undefined,
+            'open': undefined,
+            'previousClose': undefined,
+            'change': undefined,
+            'average': undefined,
+            'info': tickerData,
+        };
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchOrderBook () requires a symbol argument');
+        }
+        await this.loadMarkets ();
+        const codes = this.getIdsFromSymbol (symbol);
+        if (codes.base !== 'BTC') {
+            throw new NotSupported (this.id + ' this exchange only trades on symbols with BTC as base');
+        }
+        if (limit !== undefined) {
+            throw new NotSupported (this.id + ' fetchOrderBook () does not support a "limit" argument for this exchange');
+        }
+        let orderBookRequest = {
+            'coin': this.currencyId (codes.quote),
+            'method': 'getorders',
+        };
+        const response = await this.publicGet (orderBookRequest);
```

Comment:
Although there are no other parameters currently, they might expand the API with more, so you can do the same technique as some of the other exchanges, where you `extend` the request with `params`

Reply:
good call, i'll add to every function that has params